### ISSUE TITLE
Psalm incorrectly reports DocblockTypeContradiction when using assertSame on integers

### ISSUE THREAD

**Issue by toaomatis** [vimeo/psalm#7669]
I have the same issue on the following lines of code in my PhpUnit tests
```php
        $this->assertSame(30, $provider->getKnownDeliveryStatusCutOffTime());
        $provider->setKnownDeliveryStatusCutOffTime(14);
        $this->assertSame(14, $provider->getKnownDeliveryStatusCutOffTime());
```
On the second `assertSame` I get the message 
```
psalm: DocblockTypeContradiction: Docblock-defined type 30 for $provider->getknowndeliverystatuscutofftime() is never int(14)
```
The test obviously succeeds. 

When I change my test to 
```php
        $this->assertSame(1234, $provider->getKnownDeliveryStatusCutOffTime());
        $provider->setKnownDeliveryStatusCutOffTime(14);
        $this->assertSame(14, $provider->getKnownDeliveryStatusCutOffTime());
```
The error changes accordingly 
```
psalm: DocblockTypeContradiction: Docblock-defined type 1234 for $provider->getknowndeliverystatuscutofftime() is never int(14)
```

Version info:
```
PHP 7.4.27 (cli)
psalm: Both 4.18.1 and 4.20.0
phpunit: 9.5.13
```

The unit I'm testing is something like 
```php
<?php

final class SapProvider
{
    public const DEFAULT_KNOWN_DELIVERY_STATUS_CUT_OFF_TIME = 30;
    private int $knownDeliveryStatusCutOffTime;

    /**
     * SapProvider constructor.
     */
    public function __construct( ) {
        $this->knownDeliveryStatusCutOffTime = self::DEFAULT_KNOWN_DELIVERY_STATUS_CUT_OFF_TIME;
    }

    /**
     * @return int
     */
    public function getKnownDeliveryStatusCutOffTime(): int
    {
        return $this->knownDeliveryStatusCutOffTime;
    }

    /**
     * @param int $knownDeliveryStatusCutOffTime
     */
    public function setKnownDeliveryStatusCutOffTime(int $knownDeliveryStatusCutOffTime): void
    {
        $this->knownDeliveryStatusCutOffTime = $knownDeliveryStatusCutOffTime;
    }
```

**Comment by toaomatis** [comment]
Not quite, but you get the gist reproduction: https://psalm.dev/r/bffcfb4a5a 
Of course there is no PhpUnit installed

**Comment by weirdan** [comment]
Do you have https://github.com/psalm/psalm-plugin-phpunit installed? Also, what does your `psalm.xml` look like?

**Comment by toaomatis** [comment]
No I have not

**Comment by weirdan** [comment]
I failed to reproduce it locally. Would you mind creating a small standalone github repo with a reproducer?

**Comment by weirdan** [comment]
Does this still happen?

**Comment by toaomatis** [comment]
I'm sorry I did not follow up on your ticket. I do not work at the company I wrote this code for anymore. So I can't verify. Will close this issue for!

**Comment by stronk7** [comment]
Hi, I just faced this when trying to cover my test directories with PSalm (I had them previously uncovered). I've tried adding both the Symfony and PHPUnit plugins, just to check if they make any difference, but they do not.

I've been able to create a super-reduced repo with a minimal PHPUnit test that reproduces the problem(s).

It's here: https://github.com/stronk7/doc_block_type_contradiction_phpunit

Not sure if you may want to reopen this or, instead, report a new issue. I'm happy either way.

Ciao :-)

PS: Note that I'm a PSalm noob, so maybe I'm missing something obvious, but cannot understand what is going there, hence reporting.
